supla-mqtt-client/src/clientcfg.c

```diff
@@ -0,0 +1,120 @@
+/*
+ Copyright (C) AC SOFTWARE SP. Z O.O.
+
+ This program is free software; you can redistribute it and/or
+ modify it under the terms of the GNU General Public License
+ as published by the Free Software Foundation; either version 2
+ of the License, or (at your option) any later version.
+
+ This program is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with this program; if not, write to the Free Software
+ Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
+ */
+
+#include <pwd.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <unistd.h>
+
+#include "clientcfg.h"
+#include "supla-client-lib/log.h"
+#include "supla-client-lib/proto.h"
+#include "supla-client-lib/tools.h"
+
+char *cfg_id_file = NULL;
+char *cfg_authkey_file = NULL;
+
+char *cfg_config_file = NULL;
+
+char cfg_client_GUID[SUPLA_GUID_SIZE];
+char cfg_client_AuthKey[SUPLA_AUTHKEY_SIZE];
+
+unsigned char clientcfg_init(int argc, char *argv[]) {
+  int a;
+  struct passwd *pw;
+  char *buffer;
+  char GUIDHEX[SUPLA_GUID_HEXSIZE + 1];
+
+  for (a = 0; a < argc; a++) {
+    if (strcmp("-config", argv[a]) == 0 && a < argc - 1) {
+      cfg_config_file = strdup(argv[a + 1]);
+      a++;
+    }
+  }
+
+  if (cfg_id_file == NULL) {
+    pw = getpwuid(getuid());  // NOLINT
+    a = strlen(pw->pw_dir) + 50;
+
+    buffer = malloc(a);
+
+    if (snprintf(buffer, a, "%s/.supla-client", pw->pw_dir) < 1) {
```

Comment:
I suggest you change supla-client to supla-mqtt-client

Reply:
done